src/apphosting/githubConnections.ts

```diff
@@ -117,21 +136,30 @@ export async function linkGitHubRepository(
     installationId = await promptGitHubInstallation(projectId, location, oauthConn);
   }
 
-  let connectionMatchingInstallation = await getConnectionForInstallation(
+  const connection = await createFullyInstalledConnection(
     projectId,
     location,
+    connectionId,
+    oauthConn,
     installationId,
   );
 
-  if (!connectionMatchingInstallation) {
-    connectionMatchingInstallation = await createFullyInstalledConnection(
-      projectId,
-      location,
-      generateConnectionId(),
-      oauthConn,
-      installationId,
-    );
-  }
+  return connection;
+}
+
+/**
+ * Prompts the user to link their backend to a GitHub repository.
+ */
+export async function linkGitHubRepository(
+  projectId: string,
+  location: string,
+  createConnectionId: string | null,
+): Promise<devConnect.GitRepositoryLink> {
+  const connectionMatchingInstallation = await createGitHubConnection(
```

Comment:
Nit: The variable name `connectionMatchingInstallation` doesn't really make sense anymore.

Reply:
done